daemon/daemon.go

```diff
@@ -713,8 +726,25 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 		return nil, err
 	}
 
-	logrus.Debug("Creating images graph")
-	g, err := graph.NewGraph(filepath.Join(config.Root, "graph"), d.driver, uidMaps, gidMaps)
+	imageRoot := filepath.Join(config.Root, "image", d.driver.String())
+	if err := os.MkdirAll(filepath.Join(imageRoot, "layerdb"), 0600); err != nil {
+		return nil, err
+	}
+	fms := layer.NewFileMetadataStore(filepath.Join(imageRoot, "layerdb"))
```

Comment:
`NewFileMetadataStore` requires root to be created by the client, does not init itself.
`NewFSStoreBackend` does init in constructor, may error.
`NewFSMetadataStore` doesn't init, may fail on first insertion.

We should make them all use a similar pattern. Also maybe combine the names, at least trivial stuff like `File` vs `FS`.